include/fmt/format-inl.h

```diff
@@ -47,7 +48,11 @@ namespace internal {
 
 FMT_FUNC void assert_fail(const char* file, int line, const char* message) {
   print(stderr, "{}:{}: assertion failed: {}", file, line, message);
+#if defined(__clang__) && defined(__CUDA__) && defined(__CUDA_ARCH__)
+  std::terminate();
```

Comment:
I think we can use `std::terminate` unconditionally but please add a short comment clarifying that it was chosen instead of `abort` because of CUDA.